#  Copyright 2024-2025 Dataport. All rights reserved. Developed as part of the POSSIBLE project.
#
#  Licensed under the Apache License, Version 2.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.

diff --git a/README.md b/README.md
index 6c118d7..a8e707e 100644
--- a/README.md
+++ b/README.md
@@ -196,6 +196,7 @@ The SD Creation Wizard supports many features from the [W3C Shapes Constraint La
 - `sh:name` and `sh:description`: Human-readable text and description in natural language. Multiple languages are supported.
 - `skos:example`: List one or more example values to make field completion more convenient for end users.
 - `sh:or`: Basic support for logical operators.
+- `sh:pattern`: Support for regex patterns.
 
 
 
diff --git a/src/main/java/eu/gaiax/sdcreationwizard/api/ConversionService.java b/src/main/java/eu/gaiax/sdcreationwizard/api/ConversionService.java
index 6f40544..380805d 100644
--- a/src/main/java/eu/gaiax/sdcreationwizard/api/ConversionService.java
+++ b/src/main/java/eu/gaiax/sdcreationwizard/api/ConversionService.java
@@ -220,6 +220,11 @@ public class ConversionService {
                 validations.add(new ConstraintOption(SH.group.getLocalName(), group));
             }
 
+            var pattern = readStringProperty(propertyShape, SH.pattern);
+            if (pattern != null) {
+                validations.add(new ConstraintOption(SH.pattern.getLocalName(), pattern));
+            }
+
             //readStringProperty(propertyShape, SH.description)
             // reads string property
             Map<String, String> description = readMultiLanguageDescriptionProperty(propertyShape, SH.description);
